src/utils/myForestV2Utils.ts

```diff
@@ -0,0 +1,122 @@
+import { ProjectPurposeTypes, UnitTypes } from '@planet-sdk/common';
+import { ExtractedData } from '../features/user/MFV2/ContributionsMap/Markers/ClusterMarker';
+import themeProperties from '../theme/themeProperties';
+import { PointFeature, AnyProps } from 'supercluster';
+
+export type Accumulator = {
+  maxContributionCount: number;
+  maxContributingObject: ExtractedData | null;
+};
+
+export const getColor = (purpose: ProjectPurposeTypes, unitType: UnitTypes) => {
+  const { primaryDarkColorX, electricPurpleColor, mediumBlueColor } =
+    themeProperties;
+  if (unitType === 'm2' && purpose === 'trees') {
+    return electricPurpleColor;
+  } else if (purpose === 'conservation') {
+    return mediumBlueColor;
+  } else {
+    return primaryDarkColorX;
+  }
+};
+
+export const getClusterMarkerColors = (
+  maxContributingProject: ExtractedData | null,
+  remainingProjects: ExtractedData[]
+) => {
+  if (maxContributingProject) {
+    const { purpose, unitType } = maxContributingProject;
+    const mainProjectColor = getColor(purpose, unitType) ?? '';
+    let tertiaryProjectColor = '',
+      secondaryProjectColor = '';
+    const length = remainingProjects.length;
+    if (length === 0) {
+      tertiaryProjectColor = secondaryProjectColor = mainProjectColor;
+    } else if (length === 1) {
+      tertiaryProjectColor = getColor(
+        remainingProjects[0]?.purpose,
+        remainingProjects[0]?.unitType
+      );
+      secondaryProjectColor = mainProjectColor;
+    } else {
+      tertiaryProjectColor = getColor(
+        remainingProjects[0]?.purpose,
+        remainingProjects[0]?.unitType
+      );
+      secondaryProjectColor = getColor(
+        remainingProjects[1]?.purpose,
+        remainingProjects[1]?.unitType
+      );
+    }
+
+    return { tertiaryProjectColor, secondaryProjectColor, mainProjectColor };
+  }
+  return {
+    tertiaryProjectColor: '',
+    secondaryProjectColor: '',
+    mainProjectColor: '',
+  };
+};
+
+export const extractAndClassifyProjectData = (
+  clusterChildren: PointFeature<AnyProps>[] | undefined
+) => {
+  const extractedData: ExtractedData[] = [];
```

Comment:
`extractedData` - what is the purpose? Name accordingly. Also, it shouldn't be given the same name as a single piece of `ExtractedData` - there isn't a way to know that this is a list from the name.